values.yaml

```diff
@@ -63,10 +63,25 @@ environmentSecrets: []
   #     name: retool-github-app-private-key
   #     key: private-key
 
-# Support for external secrets https://github.com/godaddy/kubernetes-external-secrets
+# Enables support for the legacy external secrets (enabled) and the modern External Secrets Operator (externalSecretsOperator.enabled).
+# These are mutually exclusive as both enable reading in environments variables via External Secrets.
 externalSecrets:
+  # Support for legacy external secrets, note this is deprecated in favour of External Secrets Operator: https://github.com/godaddy/kubernetes-external-secrets
+  # This mode only allows a single secret name to be provided.
   enabled: false
   name: retool-config
+  # Support for External Secrets Operator: https://github.com/external-secrets/external-secrets
+  externalSecretsOperator:
```

Comment:
do you think it might be confusing to some users that while this is nested inside `externalSecrets`, if they prefer using this approach they should be setting `enabled: false` in `externalSecrets` and the nested `enabled` below to `true`?

Reply:
I think the descriptions provided in the values.yaml file clearly explaining they are mutually exclusive helps clarify that.

What would help even more so is to provide examples in the retool documentation.

I don't see a way where maintaining backwards compatibility is possible without needing to set "enabled: false" for the legacy mode. As even if "externalSecretsOperator" was not nested the boolean for enabled will still be the controlling factor for mutual exclusion.

I think a future PR can deprecate the legacy mode as a potential breaking change in favour of having the modern ESO. WDYT?

Reply:
ok makes sense!